### DISCUSSION TITLE
Keka CLI

### DISCUSSION THREAD

**Discussion by Inkoming** [aonez/Keka#1455] [Q&A] [answered]
Hello, Keka iOS & macOS (23.01) user here.
I need to use the CLI, and the help doesn't look to be fully operational.

Examples:

`-sa{a|e|s} : set Archive name mode`
What does a|e|s stands for?

`-t{Type} : Set type of archive`
Okay, what sort of types?

Usecase:

`keka 7zz a -p"mypassword" -mx=9 -mhe=on secops.7z SecOPS/*`
Is `-mhe=on` required if I need to encrypt the filenames?
Could we get a fully documented CLI help man or something please?

**Comment by aonez** [comment, marked as answer]
Hi @Inkoming!

Note this only works with the macOS version. You can check mostly all `7zz` switches here:
https://7-zip.opensource.jp/chm/cmdline/switches/index.htm

In your question:

- [-sa](https://7-zip.opensource.jp/chm/cmdline/switches/sa.htm)
- [-t](https://7-zip.opensource.jp/chm/cmdline/switches/type.htm)

**Comment by aonez** [comment]
> Is `-mhe=on` required if I need to encrypt the filenames?

Sorry I forgot this part. This is optional and if enabled encrypts all filenames. Without this option enabled all data is encrypted but you can see al filenames.

**Comment by Inkoming** [comment]
Thanks aonez!
I can't find anything about `-mhe=on` in the 7-zip 7zz switches link you provided.
Is there a full documentation somewhere?

**Reply by aonez** [reply]
You have it on the `-m` switch, the `he` option: https://7-zip.opensource.jp/chm/cmdline/switches/method.htm#HeaderEncrypt

This website is somewhat the best documentation of 7Z that I've found.
